lib/regular_expression/parser.y

```diff
@@ -66,30 +72,46 @@ rule
 
   character_group_item:
     CHAR_CLASS
+    { result = RegularExpression::AST::CharacterClass.new(val[0]) }
     | CHAR DASH CHAR
     { result = RegularExpression::AST::CharacterRange.new(val[0], val[2]) }
     | CHAR
     { result = RegularExpression::AST::Character.new(val[0]) }
+    | COMMA
+    { result = RegularExpression::AST::Character.new(val[0]) }
+    | DIGIT
+    { result = RegularExpression::AST::Character.new(val[0].to_s) }
```

Comment:
I don't think this `to_s` is necessary since you haven't called `to_i` previously